### PR TITLE
feat(native/doc): first understanding

### PR DESCRIPTION
**Pull request by PsychoShock**
This should make users know what are natives (basic understanding)

### CODE REVIEW

File: content/docs/scripting-manual/introduction/about-native-functions.md [hunk 1]
```diff
@@ -1,22 +1,100 @@
...
+FiveM provides comprehensive documentation for native functions, which can be found on our official website. This documentation is essential for understanding what each native function does, how to use it, and what parameters it requires.
 
-- Explain what natives are
-- Where can you find them
-- How to use them
--->
+- **FiveM Native Reference**: The primary resource for native functions is the [FiveM Native Reference](https://docs.fivem.net/natives/). This reference includes a searchable database of all available native functions, along with detailed descriptions, parameter lists, and usage examples.
 
-[natives-doc]: https://runtime.fivem.net/doc/reference.html
+- **FiveM Community Forums**: The [FiveM forums](https://forum.fivem.net/) are another valuable resource. Here, you can find discussions, tutorials, and examples from other developers who share their experiences and solutions related to using native functions.
```
Comment (AvarianKnight) [inline content/docs/scripting-manual/introduction/about-native-functions.md:18]: This should be just `Community Forums` and `[Cfx.re Forums](https://forum.cfx.re/)`

File: content/docs/scripting-manual/introduction/about-native-functions.md [hunk 2]
```diff
@@ -1,22 +1,100 @@
 ---
-title: About native functions
-weight: 413
+title: Understanding and Using Native Functions in FiveM
```
Comment (AvarianKnight) [inline content/docs/scripting-manual/introduction/about-native-functions.md:2]: This document shouldn't be game specific they work the same across different games (GTA5, RDR2, GTA4)

Maybe something like `Understanding and Using Natives on supported platforms`

File: content/docs/scripting-manual/introduction/about-native-functions.md [hunk 3]
```diff
@@ -1,22 +1,100 @@
...
+weight: 1300
 ---
 
-This article is under development.
-The full list of native functions can be found [here][natives-doc].
+In FiveM, you have access to what are called "native functions" or simply "natives". These functions interact with the game directly, allowing you to perform a wide variety of actions within the game environment. This guide will explain what natives are, where you can find them, and how to use them in your scripts.
```
Comment (AvarianKnight) [inline content/docs/scripting-manual/introduction/about-native-functions.md:6]: Same note as above

Rewording could be:
`In the games supported by the platform,`

File: content/docs/scripting-manual/introduction/about-native-functions.md [hunk 4]
```diff
@@ -1,22 +1,100 @@
...
 
-<!-- TODO:
-Native functions, or just 'natives', are functions that do fucking shit wtf how do i possibly explain this
+These functions are called "native" because they are built into the game engine itself. They provide a powerful interface for developers to interact with the underlying game code. Natives can perform a wide range of tasks, from simple operations like printing messages to the console, to complex actions such as manipulating player physics or creating game entities.
 
-In FiveM, you have access to what is called 'native functions'. These functions interact with the game, and only work on
-the client. It's a huge list. Very huge. The biggest you've ever seen. And it's great. To understand them you must have
-a very high IQ, like me.
+## Where Can You Find Native Functions?
+FiveM provides comprehensive documentation for native functions, which can be found on our official website. This documentation is essential for understanding what each native function does, how to use it, and what parameters it requires.
```
Comment (AvarianKnight) [inline content/docs/scripting-manual/introduction/about-native-functions.md:14]: `FiveM` -> `Cfx`

File: content/docs/scripting-manual/introduction/about-native-functions.md [hunk 5]
```diff
@@ -1,22 +1,100 @@
...
 
-- Explain what natives are
-- Where can you find them
-- How to use them
--->
+- **FiveM Native Reference**: The primary resource for native functions is the [FiveM Native Reference](https://docs.fivem.net/natives/). This reference includes a searchable database of all available native functions, along with detailed descriptions, parameter lists, and usage examples.
```
Comment (AvarianKnight) [inline content/docs/scripting-manual/introduction/about-native-functions.md:16]: Might want to include RedM native references too even if they're unofficial 

https://redm.disquse.me/natives/

File: content/docs/scripting-manual/introduction/about-native-functions.md [hunk 6]
```diff
@@ -1,22 +1,100 @@
...
 
-[natives-doc]: https://runtime.fivem.net/doc/reference.html
+- **FiveM Community Forums**: The [FiveM forums](https://forum.fivem.net/) are another valuable resource. Here, you can find discussions, tutorials, and examples from other developers who share their experiences and solutions related to using native functions.
+
+- **CitizenFX GitHub**: Users can contribute to the [CitizenFX GitHub repository](https://github.com/citizenfx/fivem) by adding descriptions, names, and new native functions. This collaborative effort helps keep the list of natives comprehensive and up-to-date, ensuring that the community benefits from the latest features and improvements.
+
+## Difference Between Client Natives and Server-Side Natives
+
+### Client Natives
```
Comment (AvarianKnight) [inline content/docs/scripting-manual/introduction/about-native-functions.md:24]: It might be worth noting here `Cfx Natives` as they are used when a resource uses `game "common"`

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md [hunk 7]
```diff
@@ -2,9 +2,104 @@
...
+For example:
+```lua
+local nativeHash = 0xDCBA2512 -- This is the hash for the native function
+Citizen.InvokeNative(nativeHash, "MY_CUSTOM_ENTRY", "Hello, FiveM!")
+```
+In this example, `0xDCBA2512` is the hash for the native function we want to call. We pass this hash along with the required parameters to `Citizen.InvokeNative`.
+
+## When to Use Citizen.InvokeNative
+
+`Citizen.InvokeNative` is typically used when you need to access functionality that is not available through Lua scripting alone. This could include advanced game mechanics, low-level engine features, or interactions with game systems that are not directly exposed to Lua.
```
Comment (AvarianKnight) [inline content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:26]: This seems redundant

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md [hunk 8]
```diff
@@ -2,9 +2,104 @@
...
+### Documented Natives
+
+For most documented native functions, you don't need to use `Citizen.InvokeNative`. Instead, you can call them directly using their provided names and parameters. This is the preferred method as it is simpler and less error-prone.
+
+### Undocumented or Incorrectly Documented Natives
+
+`Citizen.InvokeNative` is particularly useful for undocumented natives or when the documented parameters are incorrect. In such cases, you can use the native's hash directly to call the function. This requires a good understanding of what the native does and what parameters it expects.
+
+**Note:** `Citizen.InvokeNative` should primarily be used for either undocumented natives or natives with incorrect parameters defined. For documented natives, using the direct method is recommended.
+
```
Comment (AvarianKnight) [inline content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:37]: Might want to go in depth with how to get return data with natives.

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md [hunk 9]
```diff
@@ -2,9 +2,104 @@
...
+### Documented Natives
+
+For most documented native functions, you don't need to use `Citizen.InvokeNative`. Instead, you can call them directly using their provided names and parameters. This is the preferred method as it is simpler and less error-prone.
+
+### Undocumented or Incorrectly Documented Natives
+
+`Citizen.InvokeNative` is particularly useful for undocumented natives or when the documented parameters are incorrect. In such cases, you can use the native's hash directly to call the function. This requires a good understanding of what the native does and what parameters it expects.
+
+**Note:** `Citizen.InvokeNative` should primarily be used for either undocumented natives or natives with incorrect parameters defined. For documented natives, using the direct method is recommended.
```
Comment (AvarianKnight) [inline content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:36]: This should probably also state taht it should be used for natives that need to handle structs or buffers

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md [hunk 10]
```diff
@@ -2,9 +2,104 @@
...
+-- Call the native with what you have determined to be the correct parameters
+Citizen.InvokeNative(nativeHash, parameter1, parameter2, parameter3)
+```
+
+### Example Usage
+
+Here's an example of how you might use `Citizen.InvokeNative` to call a native function:
+
+```lua
+Citizen.InvokeNative(0xDCBA2512, "MY_CUSTOM_ENTRY", "Hello, FiveM!")
```
Comment (AvarianKnight) [inline content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:54]: It would probably be better to use a game native instead of a Cfx native here

```
local DOES_ENTITY_EXIST = 0x7239B21A38F536BA
Citizen.InvokeNative(DOES_ENTITY_EXIST, entityId, Citizen.ReturnResultAnyway())
```

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md [hunk 11]
```diff
@@ -2,9 +2,104 @@
...
+`Citizen.InvokeNative` is particularly useful for undocumented natives or when the documented parameters are incorrect. In such cases, you can use the native's hash directly to call the function. This requires a good understanding of what the native does and what parameters it expects.
+
+**Note:** `Citizen.InvokeNative` should primarily be used for either undocumented natives or natives with incorrect parameters defined. For documented natives, using the direct method is recommended.
+
+### Handling Incorrect Parameters
+
+Sometimes, the documentation for a native function might have incorrect or incomplete information about the parameters it expects. In these cases, you can use `Citizen.InvokeNative` to bypass the issues and call the native function directly with the parameters you have determined to be correct.
+
+For example:
+```lua
```
Comment (AvarianKnight) [inline content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:43]: This doesn't seem like a very useful example imo

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md [hunk 12]
```diff
@@ -2,9 +2,104 @@
...
+
+### Using Structures and Complex Parameters
+
+In some cases, you might need to pass complex parameters or structures. This can be done by carefully packing the parameters according to what the native function expects.
+
+```lua
+local nativeHash = 0x87654321 -- Example hash for a native that requires a complex parameter
+local param1 = 1.0
+local param2 = { x = 100.0, y = 200.0, z = 300.0 } -- Example of a vector3 structure
+Citizen.InvokeNative(nativeHash, param1, param2.x, param2.y, param2.z)
```
Comment (AvarianKnight) [inline content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:79]: This should probably be an actual example of using with a native

[SET_ENTITY_COORDS](https://docs.fivem.net/natives/?_0x06843DA7060A026B) would probably be a good native to use here

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md [hunk 13]
```diff
@@ -2,9 +2,104 @@
...
+```
+
+### Dealing with Variadic Natives
+
+Some natives accept a variable number of arguments (variadic functions). `Citizen.InvokeNative` can handle these as well, but you'll need to ensure you pass the correct number and types of parameters.
+
+```lua
+local nativeHash = 0x9ABCDEF0 -- Example hash for a variadic native
```
Comment (AvarianKnight) [inline content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:87]: You can use [VAR_STRING](https://redm.disquse.me/natives/?_0xFA925AC00EB830B9) from RedM for this example

File: content/docs/scripting-manual/introduction/about-native-functions.md (hunk 2 above)
Comment (PsychoShock) [reply content/docs/scripting-manual/introduction/about-native-functions.md:2]: It's only for `FiveM` - spoke to intern people about this.

File: content/docs/scripting-manual/introduction/about-native-functions.md (hunk 3 above)
Comment (PsychoShock) [reply content/docs/scripting-manual/introduction/about-native-functions.md:6]: Same comment as above

File: content/docs/scripting-manual/introduction/about-native-functions.md (hunk 4 above)
Comment (PsychoShock) [reply content/docs/scripting-manual/introduction/about-native-functions.md:14]: Same comment as above

File: content/docs/scripting-manual/introduction/about-native-functions.md (hunk 5 above)
Comment (PsychoShock) [reply content/docs/scripting-manual/introduction/about-native-functions.md:16]: They want to include only official links to the docs even if Disquse is responsible for RedM.

File: content/docs/scripting-manual/introduction/about-native-functions.md (hunk 1 above)
Comment (PsychoShock) [reply content/docs/scripting-manual/introduction/about-native-functions.md:18]: I agree yeah

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md (hunk 7 above)
Comment (PsychoShock) [reply content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:26]: Related to what part?

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md (hunk 13 above)
Comment (PsychoShock) [reply content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:87]: Since it's only related to FiveM - I'll keep it to FiveM only.

File: content/docs/scripting-manual/introduction/about-native-functions.md [hunk 14]
```diff
@@ -1,22 +1,100 @@
...
-- Explain what natives are
-- Where can you find them
-- How to use them
--->
+- **FiveM Native Reference**: The primary resource for native functions is the [FiveM Native Reference](https://docs.fivem.net/natives/). This reference includes a searchable database of all available native functions, along with detailed descriptions, parameter lists, and usage examples.
 
-[natives-doc]: https://runtime.fivem.net/doc/reference.html
+- **Community Forums**: The [Cfx.re Forums](https://forum.cfx.re/) are another valuable resource. Here, you can find discussions, tutorials, and examples from other developers who share their experiences and solutions related to using native functions.
+
+- **CitizenFX GitHub**: Users can contribute to the [CitizenFX GitHub repository](https://github.com/citizenfx/fivem) by adding descriptions, names, and new native functions. This collaborative effort helps keep the list of natives comprehensive and up-to-date, ensuring that the community benefits from the latest features and improvements.
```
Comment (Mathu-lmn) [inline content/docs/scripting-manual/introduction/about-native-functions.md:20]: Maybe it would be worth including this repo too ?
https://github.com/citizenfx/natives

File: content/docs/scripting-manual/introduction/about-native-functions.md [hunk 15]
```diff
@@ -1,22 +1,100 @@
...
+
+## Difference Between Client Natives and Server-Side Natives
```
Comment (Mathu-lmn) [inline content/docs/scripting-manual/introduction/about-native-functions.md:22]: It should either be "Client-Side" and "Server-Side" or "Client" and "Server"

File: content/docs/scripting-manual/introduction/about-native-functions.md [hunk 16]
```diff
@@ -1,22 +1,100 @@
...
+
+### Client Natives
+Client natives are functions that run on the player's game client. These functions are primarily used to handle tasks related to the player's immediate environment, rendering, input handling, and interactions that need to occur on the client side. Examples include manipulating the player’s view, handling UI elements, and interacting with client-specific data.
+
+### Server-Side Natives
```
Comment (Mathu-lmn) [inline content/docs/scripting-manual/introduction/about-native-functions.md:27]: Same here, you have to be consistent with naming, if you put "side" then do the same for client

File: content/docs/scripting-manual/introduction/about-native-functions.md [hunk 17]
```diff
@@ -1,22 +1,100 @@
...
+Server-side natives, on the other hand, run on the server. These functions manage game logic that needs to be controlled and validated by the server, such as player authentication, database interactions, and ensuring game state consistency. Server-side natives are crucial for maintaining the integrity and security of the game.
```
Comment (Mathu-lmn) [inline content/docs/scripting-manual/introduction/about-native-functions.md:28]: Natives don't allow database interaction AFAIK ?

File: content/docs/scripting-manual/introduction/about-native-functions.md [hunk 18]
```diff
@@ -1,22 +1,100 @@
...
+  ```lua
+  AddTextEntry(key, value)
+  ```
+
+- **GetPlayerName**: Returns the name of a player.
+  ```lua
+  local playerName = GetPlayerName(playerId)
+  ```
+
+- **DrawNotification**: Displays a notification on the screen.
```
Comment (Mathu-lmn) [inline content/docs/scripting-manual/introduction/about-native-functions.md:92]: This is not a native anymore (an old name) and I think it's not good practice to include old names in the official docs ! New name is `EndTextCommandThefeedPostTicker`.
Also this is not one of the easiest native to include in a "first understanding" docs imo. Maybe include like `GetEntityCoords` or `DeleteEntity`

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md [hunk 19]
```diff
@@ -2,9 +2,107 @@
...
+
+The `Citizen.InvokeNative` function takes the hash of a native function and any parameters that the function requires. It then calls the native function with the provided parameters, allowing you to execute code that is written in the game's native programming language (usually C++).
+
+### Using Native Hashes
+
+Every native function has a unique hash, which is a numeric identifier. You can use this hash to call the native function directly with `Citizen.InvokeNative`. This is particularly useful for undocumented natives or when you encounter issues with the documented parameters.
+
+For example:
+```lua
+local nativeHash = 0xDCBA2512 -- This is the hash for the native function
```
Comment (Mathu-lmn) [inline content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:19]: Are you sure about the hash there ? I have `0x32CA01C3`

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md (hunk 7 above)
Comment (Mathu-lmn) [reply content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:26]: I believe related to this part : What is Citizen.InvokeNative?

**Review by Mathu-lmn — CHANGES_REQUESTED** [review]
Good first understanding docs except a few changes that can be done to improve it !

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md (hunk 19 above)
Comment (PsychoShock) [reply content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:19]: Yes good catch, thanks.

File: content/docs/scripting-manual/introduction/about-native-functions.md [hunk 20]
```diff
@@ -1,22 +1,100 @@
...
 
+## What Are Native Functions?
+Native functions in FiveM are predefined functions that enable interaction with the game engine. They are essential for creating and managing game mechanics, player interactions, and various other functionalities within your server. Natives are used extensively in both client-side and server-side scripts to control gameplay elements, manipulate objects, manage player states, and much more.
 
-<!-- TODO:
-Native functions, or just 'natives', are functions that do fucking shit wtf how do i possibly explain this
+These functions are called "native" because they are built into the game engine itself. They provide a powerful interface for developers to interact with the underlying game code. Natives can perform a wide range of tasks, from simple operations like printing messages to the console, to complex actions such as manipulating player physics or creating game entities.
```
Comment (TheIndra55) [inline content/docs/scripting-manual/introduction/about-native-functions.md:11]: I'd rephrase this since natives are not only implemented in the game itself, the server also implements the natives system.

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md [hunk 21]
```diff
@@ -73,10 +74,12 @@ print("Return value: " .. tostring(returnValue))
...
 ```lua
-local nativeHash = 0x87654321 -- Example hash for a native that requires a complex parameter
-local param1 = 1.0
-local param2 = { x = 100.0, y = 200.0, z = 300.0 } -- Example of a vector3 structure
-Citizen.InvokeNative(nativeHash, param1, param2.x, param2.y, param2.z)
+local entityHandle = PlayerPedId() -- Example: Get the handle of the player's ped
+local x = 100.0
+local y = 200.0
+local z = 300.0
+local success = Citizen.InvokeNative(0x06843DA7060A026B, entityHandle, x, y, z, false, false, false, false) -- SET_ENTITY_COORDS native
```
Comment (AvarianKnight) [inline content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:81]: The hash should be as a variable name so it documents itself

File: content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md [hunk 22]
```diff
@@ -73,10 +74,12 @@ print("Return value: " .. tostring(returnValue))
...
+print("SetEntityCoords success:", success)
```
Comment (AvarianKnight) [inline content/docs/scripting-reference/runtimes/lua/functions/Citizen.InvokeNative.md:82]: This native doesn't return anything on success/fail

**Review by AvarianKnight — CHANGES_REQUESTED** [review]

File: content/docs/scripting-manual/introduction/about-native-functions.md [hunk 23]
```diff
@@ -1,22 +1,105 @@
...
-- Explain what natives are
-- Where can you find them
-- How to use them
--->
+- **FiveM Natives**: All documented natives can be found on [our official website](https://docs.fivem.net/natives/).
 
-[natives-doc]: https://runtime.fivem.net/doc/reference.html
+- **Community Forums**: The [Cfx.re Forums](https://forum.cfx.re/) are another valuable resource. Here, you can find discussions, tutorials, and examples from other developers who share their experiences and solutions related to using native functions.
+
+- **CitizenFX GitHub**: Users can contribute to the [repository](https://github.com/citizenfx/fivem) by adding descriptions, names, and new native functions. This collaborative effort helps keep the list of natives comprehensive and up-to-date, ensuring that the community benefits from the latest features and improvements.
```
Comment (AvarianKnight) [inline content/docs/scripting-manual/introduction/about-native-functions.md:20]:
```suggestion
- **Native Docs Repo**: Users can contribute to the [repository](https://github.com/citizenfx/natives) by adding descriptions, names, and new native functions.
```

**Review by AvarianKnight — CHANGES_REQUESTED** [review]
